qiskit/visualization/timeline/core.py

```diff
@@ -108,6 +108,11 @@ def time_range(self) -> Tuple[int, int]:
 
         return new_t0, new_t1
 
+    @time_range.setter
+    def time_range(self, new_range: Tuple[int, int]):
+        """Update time range to draw."""
+        self._time_range = new_range
+
```

Comment:
The reason is this: https://github.com/python/mypy/issues/1465
It may be better just to leave a comment since its a mypy bug, but the change is cheap, so I just went for it (admittedly, I did it before I found out it was a bug)

Reply:
In that case, I would remove the change. The goal is not to be mypy-clean but to move towards a situation where mypy helps. I'm okay with mypy complaining for something it wont in the future (when the bug gets fixed)